scadalts-ui/src/components/amcharts/RangeChartComponent.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+	<v-app>
+		<v-row id="chart-settings">
+			<v-col cols="5">
+				<v-menu offset-y :close-on-content-click="false">
+					<template v-slot:activator="{ on }">
+						<v-text-field v-on="on" label="Start Date" :value="startDate"></v-text-field>
+					</template>
+					<v-date-picker first-day-of-week="1" v-model="startDate"> </v-date-picker>
+				</v-menu>
+			</v-col>
+			<v-col cols="5">
+				<v-menu offset-y :close-on-content-click="false">
+					<template v-slot:activator="{ on }">
+						<v-text-field v-on="on" label="End Date" :value="endDate"></v-text-field>
+					</template>
+					<v-date-picker first-day-of-week="1" v-model="endDate"> </v-date-picker>
+				</v-menu>
+			</v-col>
+			<v-col cols="2">
+				<v-btn icon @click="reload()"> 
+					<v-icon>mdi-refresh</v-icon>
+				</v-btn>
+			</v-col>
+		</v-row>
+		<v-row id="chart-container">
+			<v-col cols="12">
+				<div ref="chartReference"></div>
+			</v-col>
+		</v-row>
+	</v-app>
+</template>
+<script>
+import LineChart from './LineChart';
+
+export default {
+	name: 'RangeChartComponent',
+	props: [
+		'pointId',
+		'pointXid',
+		'color',
+		'width',
+		'height',
+		'polylineStep',
+		'rangeValue',
+		'rangeColor',
+		'rangeLabel',
+		'showScrollbarX',
+		'showScrollbarY',
+		'showLegend',
+	],
+	data() {
+		return {
+			errorMessage: undefined,
+			chartClass: undefined,
+			isExportId: false,
+			viewId: 0,
+			startDate: '',
+			endDate: '',
+		};
+	},
+	mounted() {
+		this.initConfiguration();
+		this.generateChart();
+	},
+	methods: {
+		generateChart() {
+			if (Number(this.polylineStep) > 1) {
+				LineChart.setPolylineStep(Number(this.polylineStep));
+			}
+			this.chartClass = new LineChart(this.$refs.chartReference, this.color);
+			this.chartClass.displayControls(
+				this.showScrollbarX,
+				this.showScrollbarY,
+				this.showLegend
+			);
+
+			let promises = this.loadPoints();
+
+			Promise.all(promises).then((response) => {
```

Comment:
// could this method be async?
const pointPromises = await Promise.all(this.loadPoints());
const notLoadedPoinIds = pointPromises
.filter((res) => { res !== 'done' });
this.errorMessage = `Points given with index has not been loaded! ${notLoadedPoinIds.join(", ")}`;